src/sql/base/browser/ui/selectBox/selectBox.ts

```diff
@@ -216,6 +216,14 @@ export class SelectBox extends vsSelectBox {
 		let selectOptions: SelectOptionItemSQL[] = SelectBox.createOptions(options);
 		this.populateOptionsDictionary(selectOptions);
 		super.setOptions(selectOptions, selected);
+		// Adding following since setOptions from super alone is not setting this._selectedOption
+		let selectedOptionIndex = this._optionsDictionary.get(this._selectedOption);
+		if (selectedOptionIndex === selected) {
```

Comment:
I don't think we need to do this check - we're just updating a string value so if they're the same it's not a huge deal to do that anyways.